src/modules/course/course.service.ts

```diff
@@ -53,9 +60,81 @@ export class CourseService {
     }
   }
 
-  // Delete a Course
+  // Delete a Course by Id
   async deleteCourse(courseId): Promise<any> {
     const deletedCourse = await this.CourseModel.findByIdAndRemove(courseId);
     return deletedCourse;
   }
+
+  // Create a Schedule
+  async addScheduleCourse(
+    courseId: Schema.Types.ObjectId,
+    createScheduleDto: CreateScheduleDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        const newSchedule = new this.ScheduleModel(createScheduleDto);
+        await newSchedule.save();
+        course.schedule.push(newSchedule);
+        await course.save();
+        return newSchedule;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch {
+      throw new Error('failed to add schedule');
+    }
+  }
+
+  // update a Schedule by Id
+  async updateScheduleCourse(
+    courseId: Schema.Types.ObjectId,
+    scheduleId: Schema.Types.ObjectId,
+    updateScheduleDto: UpdateScheduleDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        let updatedSchedule = null;
+        updatedSchedule = await this.ScheduleModel.findByIdAndUpdate(
+          scheduleId,
+          updateScheduleDto,
+          { new: true },
+        );
+        return updatedSchedule;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch {
+      throw new Error('failed to update schedule');
+    }
+  }
+
+  // Delete a schedule by Id
+  async deleteScheduleCourse(
+    courseId: Schema.Types.ObjectId,
+    scheduleId: Schema.Types.ObjectId,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        let deletedSchedule = null;
+        deletedSchedule = await this.ScheduleModel.findByIdAndRemove(
+          scheduleId,
+        );
+        return deletedSchedule;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch {
+      throw new Error('failed to delete schedule');
```

Comment:
Wherever there is try block and we catching error, instead of using `Error` use something like this:
```diff
+    } catch(e) {
+      throw new InternalServerErrorException(e);
+    }
```

Reply:
ok